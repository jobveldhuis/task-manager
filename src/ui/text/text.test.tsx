import { render, screen } from "@testing-library/react-native";
import { Text, variants } from "./index";

describe("Text", () => {
  it("should match snapshots for each variant", async () => {
    const snapshots = variants.map((variant) => {
      const component = (
        <Text variant={variant}>Lorem ipsum dolor sit amet</Text>
      );
      return expect(render(component).toJSON()).toMatchSnapshot();
    });

    await Promise.all(snapshots);
  });

  it("should render correctly", () => {
    const text = <Text>Hello world, this is a beautiful new day!</Text>;
    render(text);

    expect(
      screen.findByText("Hello world, this is a beautiful new day!"),
    ).toBeTruthy();
  });
});
