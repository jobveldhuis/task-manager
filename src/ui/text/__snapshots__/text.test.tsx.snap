// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshots for each variant 1`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "lineHeight": 22,
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 2`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "textDecorationLine": "line-through",
        "textDecorationStyle": "solid",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 3`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "color": "#fff",
        "fontWeight": "bold",
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 4`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "color": "#000C49",
        "fontWeight": "bold",
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 5`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "color": "#000C49",
        "fontWeight": "bold",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 6`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "500",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 7`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "500",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 8`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "fontSize": 42,
        "fontWeight": "normal",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 9`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "fontWeight": "normal",
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;

exports[`Text should match snapshots for each variant 10`] = `
<Text
  style={
    [
      {
        "color": "#000C49",
        "fontSize": 16,
        "fontWeight": "600",
      },
      {
        "color": "#fff",
        "opacity": 0.75,
      },
      undefined,
    ]
  }
>
  Lorem ipsum dolor sit amet
</Text>
`;
